File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int total_coins(float change)
{
    int cents = round(change * 100);
    int coin_count = 0;
    int balance = cents;

    if (balance >= 25)
    {
        coin_count += (balance/25);
        balance = (balance%25);
        
    }

    if (balance >= 10)
    {
        coin_count += (balance/10);
        balance = (balance%10);
    }
    if (balance >= 5)
    {
        coin_count += (balance/5);
        balance= (balance%5);
    }

    if(balance >= 1)
    {
        coin_count += (balance/1);
        balance = (balance%1);
        
    }
    
    return coin_count;
}


int main(void)
{
    float change;
    do
    {
        change = get_float("Change owned: ");
    }
    while (change <0);
    printf("%i\n",total_coins(change));
    return 0;
}

File: pset6/cash/cash.py
from cs50 import get_float

def main():
    change = get_float("Change owed: ")
    if change >0:
        calculate_change(change)

def calculate_change(change):
    change = int(100 * change)
    counter = 0

    if(change % 25 != change):
        counter += (change // 25)
        change = change - (25* (change // 25))

    if(change % 10 != change):
        counter += (change // 10)
        change = change - (10* (change // 10))

    if(change % 5 != change):
        counter += (change // 5)
        change = change - (5* (change // 5))

    if(change % 1 != change):
        counter += (change // 1)
        change = change - (1* (change // 1))

    print(counter)

main()